src/components/Renter/RenterAd.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { PropsWithChildren } from 'react';
+import { RouteComponentProps } from 'react-router';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { v4 as uuidv4 } from 'uuid';
+import { Container, CardMedia, Card, Box, Typography, SvgIcon, Tooltip, Link } from '@material-ui/core';
+import Skeleton from '@material-ui/lab/Skeleton';
+import HowToRegIcon from '@material-ui/icons/HowToReg';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+
+import { getTenant } from '~/api/coreApi';
+import messages from '~/components/Renter/messages';
+import commonMessages from '~/common/messages';
+
+import Renter = Models.Renter;
+
+interface MatchParams {
+  id: string;
+}
+
+type Props = RouteComponentProps<MatchParams>;
+
+const DESCRIPTION_LINE_HEIGHT = 24;
+
+type housingType = 'room' | 'flat' | 'house';
+
+const RenterAd: React.FunctionComponent<Props> = (props: PropsWithChildren<Props>) => {
+  const intl = useIntl();
+  const [tenantAd, setTenantAd] = React.useState<Renter | null>(null);
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+
+  const {
+    match: {
+      params: { id },
+    },
+  } = props;
+
+  const downloadTenantAd = async () => {
+    const result = await getTenant(id);
+    if (result && result.status === 200 && result.data) {
+      setTenantAd(result.data);
+    }
+    setIsLoading(false);
+  };
+
+  let budgetDiff = 0;
+  if (tenantAd) {
+    const diff = tenantAd.maxBudget - tenantAd.minBudget;
+    budgetDiff = diff > 0 ? diff : budgetDiff;
+  }
+
+  React.useEffect(() => {
+    downloadTenantAd();
+  }, []);
+
+  const housing: housingType = tenantAd?.housingType || 'flat';
+
+  return (
+    <Container id="renter-page">
+      <Box mt={2}>
+        <Typography variant="h4">{isLoading ? <Skeleton width={400} /> : tenantAd?.title}</Typography>
+      </Box>
+      <Box mt={2}>
+        <Card>
+          <Box display="flex">
+            {isLoading ? (
+              <Skeleton variant="rect" width={150} height={150} />
+            ) : (
+              <>
+                {tenantAd?.image?.url ? (
+                  <Box maxWidth={150} maxHeight={200}>
+                    <CardMedia
+                      component="img"
+                      alt="Tenant social avatar"
+                      image={tenantAd?.image?.url}
+                      title="Tenant social avatar"
+                    />
+                  </Box>
+                ) : (
+                  <AccountCircle color="disabled" className="avatar-icon" />
+                )}
+              </>
+            )}
```

Comment:
I guess we can get rid of empty brackets.
And also I would prefer not to use nested ternary if it is possible.